homeassistant/components/rainforest_eagle/sensor.py

```diff
@@ -0,0 +1,151 @@
+"""Support for the Rainforest Eagle-200 energy monitor."""
+from datetime import timedelta
+import logging
+
+from eagle200_reader import EagleReader
+from requests.exceptions import (
+    ConnectionError as ConnectError, HTTPError, Timeout)
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_IP_ADDRESS, CONF_SCAN_INTERVAL, DEVICE_CLASS_POWER,
+    ENERGY_KILO_WATT_HOUR)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+CONF_CLOUD_ID = 'cloud_id'
+CONF_INSTALL_CODE = 'install_code'
+POWER_KILO_WATT = 'kW'
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_SCAN_INTERVAL = timedelta(seconds=30)
+
+SENSORS = {
+    "instantanous_demand": (
+        "Eagle-200 Meter Power Demand", POWER_KILO_WATT),
+    "summation_delivered": (
+        "Eagle-200 Total Meter Energy Delivered",
+        ENERGY_KILO_WATT_HOUR),
+    "summation_received": (
+        "Eagle-200 Total Meter Energy Received",
+        ENERGY_KILO_WATT_HOUR),
+    "summation_total": (
+        "Eagle-200 Net Meter Energy (Delivered minus Received)",
+        ENERGY_KILO_WATT_HOUR)
+        }
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_IP_ADDRESS): cv.string,
+    vol.Required(CONF_CLOUD_ID): cv.string,
+    vol.Required(CONF_INSTALL_CODE): cv.string
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Create the Eagle-200 sensor."""
+    ip_address = config[CONF_IP_ADDRESS]
+    cloud_id = config[CONF_CLOUD_ID]
+    install_code = config[CONF_INSTALL_CODE]
+    interval = config.get(CONF_SCAN_INTERVAL)
```

Comment:
We can remove this option. It's already available by default.

Reply:
Removed option